src/Chapter2.hs

```diff
@@ -336,7 +359,9 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList i1 i2 ls
+  | (i1 < 0) || (i2 < 0) = []
+  | otherwise = take (i2 - i1 + 1) $ drop i1 ls
```

Comment:
Your implementation is correct when the right bound is lower than the left. But I think you can implement it more efficiently by adding a separate case to your check :slightly_smiling_face:

Reply:
You're totally right, thanks!